### PR TITLE
Contact Support: Configure Chat Widget Design and Presentation logic

### LINKED ISSUE #21275: Contact Support: Configure Chat Widget Design Elements and Colors
DocsBot.Ai chat widget has options, that can be configured to suit project needs.

In this task configure chat widget design elements and colors

### PR DESCRIPTION
**Pull request by staskus**
Fixes #21275

Configure chat widget design based on Android https://github.com/wordpress-mobile/WordPress-Android/pull/18930 and build upon that implementation to address some of the issues.

In the scope of this PR:
- Hide close button
- Hide the top header
- Hide floating button
- Make sure chatbot always appears
- Make sure chatbot history is always reset
- Set color to green to match Android

Other configuration will be done in the scope of other tasks:
- https://github.com/wordpress-mobile/WordPress-iOS/issues/21316
- https://github.com/wordpress-mobile/WordPress-iOS/issues/21315 

### Implementation

I took Android's implementation https://github.com/wordpress-mobile/WordPress-Android/pull/18930 and looked for alternative solutions to make sure the chatbot always appears after opening Contact Support.

The main issue as I found is us calling `DocsBot.open` and other docs-bot-related methods before `DocsBot` is properly loaded. It seems not enough to wait for `DocsBot.init` promise to complete, and even not enough to wait for `#docsbotai-root` to be mounted. We need to wait for `shadowRoot` element to be mounted and then we can perform actions on `DocsBot`. 

```mermaid
graph LR
    A[Observe #docsbotai-root` loaded] --> B[Observe ShadowRoot loaded]
    B --> F[Inject CSS that modifies docs bot UI]
    B -->H[Remove history]
    B --> E[Open DocsBot]
```

I have limited expertise in these solutions and relied on the solutions I found on the web. Based on my testing this solution looks to be performing better and not rely on timers to perform these actions. There's still danger in elements changing in ChatBot which would not allow us to hide header or close button but the main functionality of opening docs bot should work.

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/348de7f4-f079-4f8a-a84b-8b6b44270591

### To test:

2. Open Jetpack
3. Enable Contact Support via DocsBot feature flag
4. Open Help&Support -> Contact Support
5. Confirm chat appears automatically
6. Confirm header, close button, and floating chat button are not visible
7. Interact with the chat, come back, and return
8. Confirm chat history is reset
9. Confirm chat color is green

## Regression Notes
1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

None

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by guarani** [comment]
Nice work updating the UI! I tested it today and it's loading consistently. 

Could CSS be used for hiding the elements?

**Comment by staskus** [comment]
> Could CSS be used for hiding the elements?

@guarani, I can try, it could be a safer option 👍

**Comment by staskus** [comment]
> > Could CSS be used for hiding the elements?
> 
> @guarani, I can try, it could be a safer option 👍

@guarani, yes, I remembered now. The problem with all the tweaking comes from all these elements being part of the shadow DOM. It means that the traditional way of overriding CSS styles does not work for shadow DOM elements. 

However, we can append our own internal `css` to `shadowRoot` which can then override element styles.

You can check the latest commit https://github.com/wordpress-mobile/WordPress-iOS/pull/21314/commits/f866e31476da41c5dce4e3c717b38c11ed3fc3aa

I also improved how we reference the elements, searching the close header and refresh button by class name which should help avoid hiding unnecessary items. 

Both solutions have similar amounts of complexity. Essentially we do the same thing, we either:
- Wait for shadowRoot to be loaded to hide elements through js
- Wait for shadowRoot to be loaded to hide elements through our internal CSS

**Update:**
A plus for the CSS solution since it allows to make further improvements more easily, such as allowing the chat to appear full screen on horizontal orientations and larger screens: Contact Support: Configure chat widget for iPad and horizontal orientations https://github.com/wordpress-mobile/WordPress-iOS/pull/21330

File: WordPress/Classes/ViewRelated/Support/SupportChatBot/support_chat_widget.js [hunk 1]
```diff
@@ -42,3 +42,84 @@ DocsBotAI.init = function (c) {
...
+  var chatBotSelector = "#docsbotai-root";
+
+  // Begin observation once chat bot is mounted
+  onElementMounted(chatBotSelector, document, function (element) {
+    waitForShadowRoot(element, function (shadowRoot) {
+      resetDocsBotConversation();
+      openDocsBot();
+
+      // Inject a custom css to hide unnecessary elements
+      shadowRoot.innerHTML += `<link rel="stylesheet" href="support_chat_widget.css" />`;
```
Comment (guarani) [inline WordPress/Classes/ViewRelated/Support/SupportChatBot/support_chat_widget.js]: nitpick: Setting `innerHTML` removes existing event listeners, see [Appending HTML to an element](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#appending_html_to_an_element). Would it be better to use `appendChild` to add the link element as shown in [this example](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM#internal_versus_external_styles)?

Comment (guarani) [reply WordPress/Classes/ViewRelated/Support/SupportChatBot/support_chat_widget.js]: I tested this and it's working:
```
      const linkElem = document.createElement("link");
      linkElem.setAttribute("rel", "stylesheet");
      linkElem.setAttribute("href", "support_chat_widget.css");
      shadowRoot.appendChild(linkElem);
```
And to clarify, this isn't a blocker. While `innerHTML` could affect event handlers, I haven't seen issues as a result of its use.

**Review by guarani — APPROVED** [review]
Thanks for this PR! 
Tested:
- the hiding of the header/close button and chat button 
- Chat cleared after each interaction
- injection of DocBot key 

About the shadow DOM API that DocsBot uses: I haven't worked with this API before, it seems new but also supported by the mobile web views: https://caniuse.com/shadowdomv1 (although I haven't looked at whether our minimum versions support it on both platforms). Nice work finding a reliable way of waiting for elements in the shadow DOM 👍

**Review by ravishanker — APPROVED** [review]
CSS is also working, and probably a better alternative 👍

File: WordPress/Classes/ViewRelated/Support/SupportChatBot/support_chat_widget.js (hunk 1 above)
Comment (staskus) [reply WordPress/Classes/ViewRelated/Support/SupportChatBot/support_chat_widget.js]: Thanks, @guarani. It's more important than a nitpick, I think it's worth making this change just to be safe! 👍

**Pull request merged**